src/Microsoft.DotNet.Build.CloudTestTasks/UploadClient.cs

```diff
@@ -0,0 +1,196 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="UploadClient.cs" company="">
+//   
+// </copyright>
+// <summary>
+//   The upload client.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    #region
+
+    using System;
+    using System.Collections;
+    using System.Globalization;
+    using System.IO;
+    using System.Net;
+    using System.Text;
+    using Microsoft.Build.Utilities;
+
+    #endregion
+
+    /// <summary>
+    /// The upload client.
+    /// </summary>
+    public class UploadClient
+    {
+        private TaskLoggingHelper log;
+
+        public UploadClient(TaskLoggingHelper log)
+        {
+            this.log = log;
+        }
+
+        /// <summary>
+        /// The encode block ids.
+        /// </summary>
+        /// <param name="numberOfBlocks">
+        /// The number of blocks.
+        /// </param>
+        /// <param name="lengthOfId">
+        /// The length of id.
+        /// </param>
+        /// <returns>
+        /// The <see cref="string"/>.
+        /// </returns>
+        public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
+        {
+            var numberOfBlocksString = numberOfBlocks.ToString("D" + lengthOfId);
+
+            var bytes = Encoding.UTF8.GetBytes(numberOfBlocksString);
+            return Convert.ToBase64String(bytes);
+        }
+
+        /// <summary>
+        /// The upload block blob.
+        /// </summary>
+        /// <param name="AccountName">
+        /// The account name.
+        /// </param>
+        /// <param name="AccountKey">
+        /// The account key.
+        /// </param>
+        /// <param name="ContainerName">
+        /// The container name.
+        /// </param>
+        /// <param name="filePath">
+        /// The file path.
+        /// </param>
+        /// <param name="destinationBlob">
+        /// The destination blob.
+        /// </param>
+        public void UploadBlockBlob(
```

Comment:
Since this method is doing a _lot_ of web and file work, it would be appropriate to make it UploadBlockBlobAsync, and return a Task, so that users don't their threads blocked if they want to do some other stuff while all this IO is taking place.